### PR TITLE
made a few misc cleanup changes and added a script to re-run the processors for archived years.

### PR DESCRIPTION
**Pull request by Peedee2002**
externalised a new endpoint for legacy terms to make it possible to fetch the legacy courses for any year (including fetching the terms)

### CODE REVIEW

File: backend/data/final_data/archive/runprocessors_archive.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+"""
+The driver for our procsesors. Provide the relevant command line arguments
```
Comment (joooooooooooooooooooooooooooooooooooosh) [inline backend/data/final_data/archive/runprocessors_archive.py]: literally unreviewable 0/10

File: backend/server/routers/programs.py [hunk 2]
```diff
@@ -135,22 +135,25 @@ def getMinors(programCode: str):
 
     return {"minors": minrs}
 
-def addSubgroupContainer(structure, type, container, exceptions):
+def convertSubgroupObjectToCoursesDict(object: str, description: str|list[str]) -> dict[str, str]:
```
Comment (joooooooooooooooooooooooooooooooooooosh) [inline backend/server/routers/programs.py:138]: fairly self explanatory but may as well include a docstring

**Review by joooooooooooooooooooooooooooooooooooosh — CHANGES_REQUESTED** [review]

File: backend/data/final_data/archive/runprocessors_archive.py (hunk 1 above)
Comment (Peedee2002) [reply backend/data/final_data/archive/runprocessors_archive.py]: oops yea i should have rewritten this doctstring lol

**Review by hdphuong — APPROVED** [review]

File: backend/data/scrapers/payload.py [hunk 3]
```diff
@@ -12,7 +12,7 @@
     "content-type": "application/json",
 }
 
-def create_payload(size, content_type):
+def create_payload(size, content_type, year):
```
Comment (imagine-hussain) [inline backend/data/scrapers/payload.py]: Should `year` here have a default value set to `LIVE_YEAR`?

**Review by imagine-hussain — CHANGES_REQUESTED** [review]

File: backend/data/scrapers/payload.py (hunk 3 above)
Comment (Peedee2002) [reply backend/data/scrapers/payload.py]: Yup you're right

**Review by banaboi — APPROVED** [review]

**Review by imagine-hussain — APPROVED** [review]

**Review by joooooooooooooooooooooooooooooooooooosh — APPROVED** [review]

**Pull request merged**